base/acme/src/main/java/org/dogtagpki/acme/server/ACMEFinalizeOrderService.java

```diff
@@ -97,14 +98,14 @@ public Response handlePOST(@PathParam("id") String orderID, JWS jws) throws Exce
         //    in [RFC3339].  This field is REQUIRED for objects with "pending"
         //    or "valid" in the status field.
 
-        // set valid order to expire in 30 minutes
-        // TODO: make it configurable
-
-        long currentTime = System.currentTimeMillis();
-        Date expirationTime = new Date(currentTime + 30 * 60 * 1000);
-
         order.setStatus("valid");
-        order.setExpirationTime(expirationTime);
+
+        ACMEValidity validity = engine.getConfig().getValidity("validOrder");
```

Comment:
Is now a good time to ask... do we need to have a separate validity period for after an order becomes valid?  Things should happen pretty quickly, I don't really think we need to reset the expiry after order transitions to `valid`.  What do you think @edewata @cipherboy @SilleBille ?

Reply:
In case the server cannot start the challenge validation immediately, the order could get stuck in pending state for a long time. I think the validity of a valid order should not depend on how long the order has been pending.